src/platform/testing/e2e/form-tester/form-filler.js

```diff
@@ -304,6 +304,18 @@ const removeForeseeOverlay = async page => searchAndDestroy(page, '.__acs');
  *  page.
  */
 const fillForm = async (page, testData, testConfig, log) => {
+  const runHook = async hook => {
+    if (typeof hook !== 'function') {
+      throw new Error(
+        `Bad testConfig: Page hook for ${page.url()} is not a function`,
+      );
+    }
+    const retVal = hook(page, testData, testConfig, log);
+    if (retVal instanceof Promise) {
+      await retVal;
```

Comment:
I think you could just return this without checking if it's a promise type. I think async functions wrap non-Promise return values into a resolved promise implicitly.